Add empty WordPressData target + custom test plan

mokagio opened the pull request:
To avoid long lived branches like #24242 and integration issues as the work on Keystone advances, here's a PR that at leasts adds the scaffolding for WordPressData so we'll be able to add sources and tests there in the future.

Part of #24165

mokagio committed:
Add empty WordPressData target + custom test plan

mokagio committed:
Do not use a host app to run WordPressData tests

mokagio committed:
Move WordPressData sources to Sources/WordPressData

mokagio committed:
Move WordPressDataTests to Tests folder

mokagio committed:
Remove WordPressData the Swift package as an Xcode target dependency

mokagio committed:
Add dummy type to verify setup

mokagio committed:
Use the same settings as Keystone for WordPressData

mokagio committed:
fix tests

mokagio committed:
Use the same build settings as Authenticator in WordPressDataTests

mokagio committed:
Remove WordPressData from WordPress' dependencies

It's not time for that yet

dangermattic commented:
<!--
  0 Errors
  3 Warnings: <code>Modules/Package.swift</c...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Adjust linking so that WordPress, Jetpack, and the tests build

The issue is that I named the framework and the package both
WordPressData and that we already have existing `import`s of it in
production and test code.

To keep the diff simple, I decided to leave those imports in place and
simply:

- Remove the WordPressData package (empty) dependency
- Add explicit WordPressDataObjc package (empty) dependency
- Add WordPressData framework (empty) dependency

This sorts the compilation and running locally.

As we'll move files to WordPressData and/or away from Keystone, we might
need to change this approach.

mokagio committed:
Explicitly set code signing for WordPressData

See how this build
https://buildkite.com/automattic/wordpress-ios/builds/26971 failed

mokagio committed:
Set provisioning style as manual in WordPressData
